api/models/site.js

```diff
@@ -42,6 +68,16 @@ function toJSON() {
   object.createdAt = object.createdAt.toISOString();
   object.updatedAt = object.updatedAt.toISOString();
 
+  // prevent any invalid domain or demoDomain URLs from being returned
+  // or used in the creation of viewLink or demoViewLink
+  if (!isEmptyOrUrl(object.domain, false)) {
+    object.domain = null;
+  }
+
+  if (!isEmptyOrUrl(object.demoDomain, false)) {
```

Comment:
Yeah, I think that would be better, rather than complicating the logic of a security-relevant bug fix to avoid touching it. We could definitely get bugs that require us to migrate production data, so this just seems like something that falls into that class.